app/controllers/concerns/course_fetch_concern.rb

```diff
@@ -0,0 +1,41 @@
+module CourseFetchConcern
```

Comment:
Probably rename this to `CourseFetcher` or an alternative as the concern is already implied from the path